### ISSUE TITLE
Using fromPromise in effects where the promise rejects - catchError doesn't work and promise error remains uncaught

### ISSUE THREAD

**Issue by fmflame** [ngrx/platform#848]
## I'm submitting a...
<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Feature request
[ ] Documentation issue or request
</code></pre>

## What is the current behavior?
Using fromPromise in effects where the promise rejects - catchError doesn't work and promise error remains uncaught.

## Expected behavior:
I expect if everything is ok the SuccessAction to be dispatched and if the promise rejects ErrorAction to be dispatched.

## Minimal reproduction of the problem with instructions:
Just create an effect like this:
```
@Effect()
    public someAction: Observable<Action> = this.actions.pipe(
        ofType(SomeAction.TYPE),
        mergeMap((action) => {
            let p = new Promise((resolve, reject) => { reject({ status: 401 }); });
            return fromPromise(p).pipe(
                map((response) => {
                    return new SuccessAction();
                }),
                catchError((err) => {
                    return of(new ErrorAction());
                })
            );
        })
    );
```

## Version of affected browser(s),operating system(s), npm, node and ngrx:
Windows 10, Chrome, npm 5.4.2, node 8.6.0, ngrx 4.1.1 but also tested with latest, rxjs 5.5.6, angular 5.2.5

## Other information:
I am not sure If my coding is correct but I've tried lots of things and nothing works. The same example as above but not in effect where I subscribe to the observable works, but this doesn't. 
So this might not be a bug, but in any way please let me know how to make this work.

Thank you!

P.S.
I also posted this problem on stackoverflow here: https://stackoverflow.com/questions/48907600/in-angular-5-ngrx-5-rxjs-5-5-6-using-frompromise-in-effects-where-the-promise

**Comment by brandonroberts** [comment]
Can you post an actual reproduction of this issue? I pasted your code in the example app and ran it with no issues.

**Comment by fmflame** [comment]
Ok so I started recreating an example for your from a basic example and then upgraded from there and I actually found out the issue was in one of my resolvers which returns Observable.throw which is not handled anywhere and results in this promise uncaught error. Turns out to be a very easy fix and not a bug for you!
Sorry for wasting you time, I did not relate the promise uncaught with an Observable.throw at the time I wrote this.
Thank you and have a nice day! :)
